index.js

```diff
@@ -337,6 +337,14 @@ class TrezorKeyring extends EventEmitter {
     this.paths = {};
   }
 
+  setHdPath (hdPath) {
+    // Reset HDKey if the path changes
+    if (this.hdPath !== hdPath) {
+      this.hdk = new HDKey()
+    }
+    this.hdPath = hdPath
```

Comment:
Did you consider resetting the other instance variables? (e.g. `account`).

Reply:
The instance variables `paths` and `unlockedAccount` are still not being reset.

Reply:
Thanks. Fixed in latest commit